Separators under headings with hexagon

msecret opened the pull request:
These were built so they could conform to many different widths
and the middle piece could be switched out. With this design, the
width of the middle piece can be changed and the lines will conform.
The hexagon class uses ems for this reason, so you can set the font
size to control it's size. This had to be done because borders can't
be set with percentages so ems are used instead. And then the size
of the hexagon is set with the parent font-size.

It can also change colors from primary to alt or any other.

Commit:
Separators under headings with hexagon

These were built so they could conform to many different widths
and the middle piece could be switched out. With this design, the
width of the middle piece can be changed and the lines will conform.
The hexagon class uses ems for this reason, so you can set the font
size to control it's size. This had to be done because borders can't
be set with percentages so ems are used instead. And then the size
of the hexagon is set with the parent font-size.

It can also change colors from primary to alt or any other.

jcscottiii commented on src/css/components/hexagon.scss:
```diff
@@ -0,0 +1,47 @@
+
+.hexagon {
+	background: $primary-color;
+  height: .65em;
```
uneven spacing between lines

jcscottiii commented on src/css/components/hexagon.scss:
```diff
@@ -0,0 +1,47 @@
+	border-bottom: .25em solid $primary-color;
+	content: "";
+	position: absolute;
+  top: -.25em;
```
same here

jcscottiii replied:
well this line has the right spacing, but not the others around it

jcscottiii commented on src/css/components/hexagon.scss:
```diff
@@ -0,0 +1,47 @@
+}
+	
+.hexagon-alt {
+	background: $alt-color;
```
spacing

jcscottiii commented on src/css/components/hexagon.scss:
```diff
@@ -0,0 +1,47 @@
+	background: $primary-color;
+  height: .65em;
+	position: relative;
+  width: 1em;
```
em -> rem?

msecret replied:
These are actually all supposed to be `em` because you can set teh font-size of the parent to control the size of the hexagon. This was done rather then using percentage units for everything because border sizes can't be defined with percentages but can be defined with `em`s.

jcscottiii commented on src/css/components/hexagon.scss:
```diff
@@ -0,0 +1,47 @@
+
+
+.hexagon:before {
+	border-left: .5em solid transparent;
```
em -> rem?

jcscottiii commented on src/css/components/hexagon.scss:
```diff
@@ -0,0 +1,47 @@
+}
+
+.hexagon:after {
+	border-left: .5em solid transparent;
```
em -> rem

jcscottiii commented on src/css/components/separator.scss:
```diff
@@ -0,0 +1,47 @@
+  flex-grow: 0;
+  flex-shrink: 1;
+  font-size: $center_piece_size;
+  margin-left: 1em;
```
em -> rem?

Commit:
Fix formatting and organization

For some reason, code had some weird tab issues, should be fixed.

Re-orged :before and :after selectors so they use the same properties
that are shared to keep code more dry.
